src/Test/Utilities/DiagnosticAnalyzerTestBase.cs

```diff
@@ -84,6 +88,21 @@ public static MetadataReference SystemThreadingTaskFacadeRef
             }
         }
 
+        protected bool PrintActualDiagnosticsOnFailure
+        {
```

Comment:
Yes, set-only properties are weird. What's going on here is clearer if you remove the backing field; then you have:
```
protected bool PrintActualDiagnosticsOnFailure { private get; set; }
```
... at which point I'd ask, what's the harm in the derived class being able to recall the value that they set? I'd just change it to
```
protected bool PrintActualDiagnosticsOnFailure { get; set; }
```